File: sdk/ocaml/lib/location_v2.ml
(** LocationV2: countries, states, cities lookup *)

(** GET /api/location_v2/countries - Return list of countries *)
let list_countries client params =
  HttpClient.get client "/api/location_v2/countries" params

(** GET /api/location_v2/{code}/state - Return states for a country code *)
let get_states client params =
  HttpClient.get client "/api/location_v2/{code}/state" params

(** GET /api/location_v2/state/{id}/city - Return cities for a state ID *)
let get_cities client params =
  HttpClient.get client "/api/location_v2/state/{id}/city" params

(** GET /api/location_v2/state/city/{name} - Return cities for a state name *)
let get_cities_by_state_name client params =
  HttpClient.get client "/api/location_v2/state/city/{name}" params

File: sdk/rust/src/location_v2.rs
//! LocationV2: countries, states, cities lookup

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct LocationV2Resource {
    http: Arc<HttpClient>,
}

impl LocationV2Resource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// GET /api/location_v2/countries - Return list of countries
    pub async fn list_countries(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/location_v2/countries", None).await
    }

    /// GET /api/location_v2/{code}/state - Return states for a country code
    pub async fn get_states(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/location_v2/{code}/state", None).await
    }

    /// GET /api/location_v2/state/{id}/city - Return cities for a state ID
    pub async fn get_cities(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/location_v2/state/{id}/city", None).await
    }

    /// GET /api/location_v2/state/city/{name} - Return cities for a state name
    pub async fn get_cities_by_state_name(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/location_v2/state/city/{name}", None).await
    }

}
